symplyphysics/laws/thermodynamics/entropy_from_statistical_weight.py

```diff
@@ -16,9 +24,186 @@
 
 law = Eq(entropy, units.boltzmann_constant * log(statistical_weight))
 
+# Derive the law from the properties of entropy
 
-def print_law() -> str:
-    return print_expression(law)
+# Let us find entropy as a function of probability of the system's state, i.e. `S = f(P)`,
+# where `S` is entropy, `P` is probability, and `f` is the functional dependency to be found.
+# This dependency must be universal for all physical systems and should only depend on the system's
+# probability.
+
+# Suppose there are two independent subsystems in states with probabilities `P_1` and `P_2`
+# respectively. If we join the two subsystems into one, the probability of the total system
+# will be `P_12`, and using their independence we have `P_12 = P_1 * P_2`.
+
+_time = symbols("time", real=True)
+_first_probability = symbols("first_probability", cls=SymFunction, positive=True)(_time)
+_second_probability = symbols("second_probability", cls=SymFunction, positive=True)(_time)
+_entropy = symbols("entropy", cls=SymFunction, real=True)
+
+# According to thermodynamics, the entropy of a complex system should equal the sum of entropies
```

Comment:
Do we have a law for this? Add TODO if we need it.

Reply:
Opened #748